### ISSUE TITLE
Significant slowdown in Keras model.fit() on simple problem when using validation data

### ISSUE THREAD

**Issue by aaustin141** [keras-team/keras#17328]
**System information**.
- Have I written custom code:  No
- OS Platform and Distribution:  Ubuntu 18.04
- TensorFlow installed from:  Binary
- TensorFlow version:  2.11.0
- Python version:  3.10.8
- Bazel version:  N/A
- GPU model and memory:  NVIDIA Quadro GV100 (32 GB)
- Exact command to reproduce:  See below.

**Describe the problem**.

Per instructions I received from @mohantym, this is a cross-post to this
repository of the issue tensorflow/tensorflow#58828, which I opened yesterday.

The issue documents a significant slowdown in training speed when using Keras
model.fit() with the validation_data parameter set.  On my hardware, I see a
slowdown of approximately 10x.  On Colab, @mohantym sees a slowdown of only
3.5x, but this still seems excessive to me given that the validation data is
only 1/8 the size of the training data.  I would not expect it to take 3.5x
time to process 12.5% more data per epoch.

(In fact, I see the same slowdown even if I reduce the validation set size to a
single example, so it's a 3.5x to 10x speed penalty for an 0.05% increase in
the amount of data to be processed.)

For further details, including code that can be used to reproduce the problem,
please see the aforementioned issue.

**Describe the current behavior**.

A 3.5x-10x slowdown during training using model.fit() with validation_data set.

**Describe the expected behavior**.

Some slowdown (validation isn't free)---but a whole lot less than 3.5x-10x.

**Standalone code to reproduce the issue**.

See the aforementioned issue.

**Source code / logs**.

See the aforementioned issue.

**Comment by sushreebarsa** [comment]
@aaustin141 Could you refer to this [thread](https://stackoverflow.com/questions/66157475/why-does-keras-training-slow-down-after-a-while) and let us know if it helps?
Thank you!

**Comment by SuryanarayanaY** [comment]
Hi @aaustin141 ,

Since Evaluation happens at the end of each epoch when we provide `validation_split` to `model.fit()` there will be some extra time overhead for this operation.Since the model you tested is very small and time consumption also very little the difference seems huge.However this difference in time will reduce with larger models.The same model mentioned by you in #58828 I tried with 1000 epochs and time taken with and without validation is 40.87 to 20.52 Seconds respectively which is 2X compared to 3.75X earlier .Same model if Eager mode disabled by using `tf.compat.v1.disable_eager_execution()` then time taken is 17.67 and 12.62 Sec respectively where the difference is very less.I believe this difference may come down with larger models with large training time. but AFAIK there is no such metrics to confirm whether `validation_split` should take only this much time.  Please refer to attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/25268dae94cfb86239374011735956f5/17328.ipynb).

Could you please confirm whether you are facing this issue with any of your specific model and if so can you confirm the training time details for it ?

Thank you!
